components/post-text/PostText.tsx

```diff
@@ -37,18 +38,32 @@ const PostText = ({ text }: IProps) => {
     );
   };
 
-  const replaceHashtags = (text: string) => {
-    return text.replace(regexExp, (match: string, hashtag: string): string => {
-      const searchKeyword = match.replace('#', '').toLowerCase();
-      // this is not jsx. it's a simple string
-      return `<a class="text-violet-600" href="/posts/hashtag/${searchKeyword}">${match}</a>`;
-    });
+  const replaceHashtags = (text: string): TrustedHTML | null => {
+    if (!text) return null;
+    return text
+      .replace(/\n/g, '<br>')
+      .replace(regexExp, (match: string, hashtag: string): string => {
+        const searchKeyword = match.replace('#', '').toLowerCase();
+        // this is not jsx. it's a simple string
+        return `<a class="text-violet-600" href="/posts/hashtag/${searchKeyword}">${match}</a>`;
+      });
   };
 
   return (
     <div className="text-slate-600 font-poppins not-italic font-medium text-lg leading-[1.40]">
-      <div dangerouslySetInnerHTML={{ __html: replaceHashtags(text) }}></div>
-      {renderHashTags(text)}
+      {(() => {
+        const html = replaceHashtags(text);
+        if (!html) return;
+        return (
+          <div
+            dangerouslySetInnerHTML={{
```

Comment:
it might be good to have a comment why we use dangerouslySetInnerHTML here

Reply:
👍🏽 , the answer (comment) is at line 47.